merger_sort.c

```diff
@@ -0,0 +1,98 @@
+#include <stdio.h>
+
+#define SIZE 8
+
+int temp[SIZE] = {0};
+
+void merge (int array[], int start_1, int end_1, int start_2, int end_2)
+{
+    int i = 0;
+    
+    // while there are elements in both subarrays 
+    while ((start_1 <= end_1) && (start_2 <= end_2))
+    {
+        // compare numbers at the start of the subarrays 
+        if (array[start_1] > array[start_2])
+        {
+            // append smaller to array 
+            temp[i] = array[start_2];
```

Comment:
`temp` is, in theory, a pointer to a global array with one `int` element. Make the size of the original array in main larger and something _**will**_ break.

You're probably looking for something along the line of
`int temp[end_2 - start_1 + 1];`